proposals/jib-cli-surface.md

```diff
@@ -0,0 +1,60 @@
+# Jib CLI Surface Specification
+
+Specification for a command structures exposed to users of the jib-cli
+
+## Usage
+`jib COMMAND [OPTIONS]`
+
+## Commands
+
+The jib cli exposes the following top level commands, mirroring what is available in the plugins
+```
+build        build to registry
+dockerBuild  build to docker daemon
+buildTar     build to a tar file
```

Comment:
Just food for thoughts:

We have had multiple instances of feedback to be able to "warm up a cache" without pushing or creating a tar. Honestly, I wish the Jib build plugins had a way. And if we support the URI style, I think there's no need to have three commands.

Related, we may considering providing an option to configure the application layer cache (the default is to use a temporary directory) and the base image cache, but I'm not sure how useful they will be in practice.

Reply:
Added a warm up cache section at the end, presumably that would be a separate command? we'd have to provide a shortcut in jib-core stepsrunner to do it though